api/src/main/java/org/pmiops/workbench/api/NotebooksController.java

```diff
@@ -70,14 +72,22 @@ private FileDetail copyNotebookImpl(
       CopyRequest copyRequest) {
     FileDetail fileDetail;
     try {
+      // Checks the new name extension to match the original from file type, add extension if
+      // needed.
+      // TODO(yonghao): Remove withNotebookExtension after UI start setting extension.
+      String newName =
+          isRMarkdownNotebook(fromNotebookName)
+              ? NotebooksService.withRMarkdownExtension(copyRequest.getNewName())
+              : NotebooksService.withJupyterNotebookExtension(copyRequest.getNewName());
+
       fileDetail =
           notebooksService.copyNotebook(
               fromWorkspaceNamespace,
               fromWorkspaceId,
-              NotebooksService.withNotebookExtension(fromNotebookName),
+              NotebooksService.withJupyterNotebookExtension(fromNotebookName),
```

Comment:
Does your comment above (about removing withNotebookExtension) apply to this line too?

Reply:
yup. also copied the same comment over